### PR TITLE
Add initial cypress tests to HUB

### PR DESCRIPTION
**Pull request by nixocio**
Add initial cypress tests to HUB

```
npm run e2e:run:hub
```

image

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -579,6 +583,41 @@ jobs:
...
+      - name: wait for API be available
+        env:
+          API_HUB_SERVER_URL: http://localhost:5001/api/automation-hub/_ui/v1/feature-flags/
+        run: |
+          MAX_ATTEMPTS=6000
+          DELAY_BETWEEN_ATTEMPTS=10
+          TIMEOUT_REACHED=false
+          http_status_code=0
+          for ((i = 0; i < MAX_ATTEMPTS; i++)); do
+            response=$(curl -s "$API_HUB_SERVER_URL")
```
Comment (himdel) [inline .github/workflows/pull-request.yml:595]:
```suggestion
            response=$(curl -s "$API_HUB_SERVER_URL" || true)
```

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -579,6 +583,41 @@ jobs:
...
+            echo "API response: $response"
+            if [ -z "$response" ]; then
+              echo "API returned an empty response. Retrying."
+              sleep "$DELAY_BETWEEN_ATTEMPTS"
+            else
+              echo "API returned a non-empty response. Checking status code."
+              http_status_code=$(curl -v -w "%{http_code}" "$API_HUB_SERVER_URL")
```
Comment (himdel) [inline .github/workflows/pull-request.yml:602]:
```suggestion
              http_status_code=$(curl -v -w "%{http_code}" "$API_HUB_SERVER_URL" || true)
```

Comment (nixocio) [reply .github/workflows/pull-request.yml]: Thanks @himdel, changes added

**Review by vidyanambiar — APPROVED** [review]
lgtm

**Pull request merged**